src/domains/misc/components/Modal/ModalProvider.tsx

```diff
@@ -89,22 +88,26 @@ export const ModalProvider = ({ children }: { children: ReactNode }) => {
   );
 };
 
-export const useModal = (modalElement: ReactElement) => {
+export const useModal = () => {
   const ctx = useContext(ModalContext);
   if (!ctx) throw new Error('useModal must be used within ModalProvider');
 
   const { mount, unmount, modals } = ctx;
-  const id = useRef(uuidv4()).current;
 
-  const open = useCallback(() => {
-    mount({ id, modal: modalElement });
-  }, [id, modalElement, mount]);
+  const [defaultId] = useState(uuidv4());
+  const currentIdRef = useRef(defaultId);
+
+  const open = useCallback((modalElement: ReactElement, options?: { idOverride?: string }) => {
+    const modalId = options?.idOverride ?? defaultId;
+    currentIdRef.current = modalId;
```

Comment:
Sorry to bother you again about it, we could do
```ts
const currentIdRef = useRef<string | undefined>(undefined);

...

currentIdRef.current = modalId ?? uuidv4();
```
and remove state completely :sweat_smile: 

WDYT? Feel free to ignore